### PR TITLE
update docs to use node 8 + yarn

### PR DESCRIPTION
**Pull request by justindowning**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,8 +34,8 @@ $ heroku plugins:install heroku-kafka
 
 For normal development, the initial setup is:
 ``` sh-session
-# ensure node 6.x is installed
-$ npm install
+# ensure node 8.x is installed
+$ yarn
```
Comment (uhoh-itsmaciek) [inline README.md]: I think this actually be `yarn --ignore-engines` (and possibly even `yarn --ignore-engines --ignore-scripts`--I think @idan told me to run that, but lately just `yarn --ignore-engines` seems to work for me). Does plain `yarn` work for you? I get:

```
maciek@mothra:~/code/herokudata$ yarn
yarn install v0.24.6
[1/4] Resolving packages...
[2/4] Fetching packages...
error urlsafe-base64@0.0.2: The engine "node" is incompatible with this module. Expected version "0.8.x".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Comment (uhoh-itsmaciek) [reply README.md]: Confirming that just `yarn --ignore-engines` works here.

Comment (justindowning) [reply README.md]: `yarn` alone worked for me.

```
$ node --version          
v8.4.0
$ yarn --version          
0.27.5
$ rm -rf node_modules     
$ yarn                    
yarn install v0.27.5
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 7.99s.
```

Comment (uhoh-itsmaciek) [reply README.md]: Weird. Any idea what's up, @fnando @idan?

Comment (idan) [reply README.md]: Uh… is "no"  a legit answer? #justyarnthings

Comment (idan) [reply README.md]: IDK, I'd add a note to include `--ignore-engines` if something gives you guff. The reason we use the flag is that some dependency of our dependencies has a bad `engines` block in their package.json, e.g. one which limits engines to some ancient version of node despite the code being perfectly fine with newer nodes. If you don't have this dep anymore, then `--ignore-engines` shouldn't be needed.

Comment (fnando) [reply README.md]: @justindowning what does `cat node_modules/urlsafe-base64/package.json` return?

Comment (justindowning) [reply README.md]: @fnando that module is not a dependency ✨

Comment (fnando) [reply README.md]: what `yarn why urlsafe-base64` returns?

Comment (justindowning) [reply README.md]:
```
~/dev/heroku/heroku-kafka-jsplugin (update_docs ✔) ᐅ yarn why urlsafe-base64
yarn why v1.0.1
[1/4] 🤔  Why do we have the module "urlsafe-base64"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 0.39s.
```

Comment (fnando) [reply README.md]: derp. Looks like @uhoh-itsmaciek ran the command on herokudata, instead of `heroku-kafka-jsplugin`. Didn't see this before.

**Pull request merged**